Source/carthage/BuildVersion.swift

```diff
@@ -5,8 +5,43 @@ import ReactiveTask
 import Result
 import Tentacle
 
+struct ExpirableVersion: Codable {
+	let version: SemanticVersion?
+	let expiredDate: Date
+
+	init(version: SemanticVersion?, shelfLife: TimeInterval) {
+		self.version = version
+		self.expiredDate = Date(timeIntervalSinceNow: shelfLife)
+	}
+
+	init?(fromJSONData data: Data?) {
+		guard let wrappedData = data, let wrapped = try? JSONDecoder().decode(ExpirableVersion.self, from: wrappedData) else {
+			return nil
+		}
+		self = wrapped
+	}
+
+	func encodedJSONData() -> Data? {
+		return try? JSONEncoder().encode(self)
+	}
```

Comment:
Instead of building this into this type, can we add a `UserDefaults` extension to decode/encode `Codable` types? That should make it easy to add some tests.